### PR TITLE
mark custom attributes

### LINKED ISSUE #9: Provide flag to mark unknown attribute
Hi, I need to check if an attribute is a known HTML attribute or not. I do it in this way:

```js
info.property === info.attribute // unknown
```
I think it would be good to provide a flag like `unknown: true`

### PR DESCRIPTION
**Pull request by StarpTech**
Fix #9 
* add new property `custom` when attribute is not part of the spec

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -228,6 +228,10 @@ Info on a property.
 *   `mustUseProperty` (`boolean`) — If a DOM is used, setting the property
     should be used for the change to take effect (this is true only for
     `'checked'`, `'multiple'`, `'muted'`, and `'selected'`)
+*   `unspecified` (`boolean`) — The property is not in the [list](#support)
```
Comment (StarpTech) [inline readme.md]: @wooorm I think `unsupported` make more sense?

File: lib/util/info.js [hunk 2]
```diff
@@ -29,6 +30,7 @@ function Info(property, attribute, mask, space) {
   mark(this, 'commaSeparated', check(mask, types.commaSeparated))
   mark(this, 'spaceSeparated', check(mask, types.spaceSeparated))
   mark(this, 'commaOrSpaceSeparated', check(mask, types.commaOrSpaceSeparated))
+  mark(this, 'unspecified', isUnspecifiedProperty(property, attribute, space))
```
Comment (wooorm) [inline lib/util/info.js]: Would it make sense to have to constructors? One that’s used by the schema‘s, one for [`find` as a fallback](https://github.com/wooorm/property-information/blob/911285c79dbf8f8eb842849536324ffa196f8c88/find.js#L31)?

File: readme.md (hunk 1 above)
Comment (wooorm) [reply readme.md]: hmm. I kinda feel like we added “support” for these unspecified properties in GH-7.

Comment (StarpTech) [reply readme.md]: Yes, but with unsupported I mean that we can't provide additional information.
The [GH-7](https://github.com/wooorm/property-information/issues/7) was about to fix normalization.

Comment (StarpTech) [reply readme.md]: That leads me to the name `untouched` because we do nothing with those properties.

File: lib/util/info.js (hunk 2 above)
Comment (StarpTech) [reply lib/util/info.js]: Could you rephrase? I don't get it.

Comment (wooorm) [reply lib/util/info.js]: Sorry. Well, `Info` is a class / constructor. We could rename it to `DefinedInfo`.
And create a new class for `UndefinedInfo`, and use that for `find` results that are not defined.
(or other names)

Comment (StarpTech) [reply lib/util/info.js]: Ah, you can want to classify it with the help of constructors. I like the idea. I will update the PR tomorrow 👍

Comment (wooorm) [reply lib/util/info.js]: I think it would be good for performance to have all “classified” properties inherit from one thing.
And on `.find()`, when there’s no result, construct something specific that has everything set to `false`, plus this new field!

Comment (StarpTech) [reply lib/util/info.js]: e.g
```
Info (attribute, property, space ...) -> DefinedInfo
     						          -> UndefinedInfo (unspecified)
```

**Comment by StarpTech** [comment]
WDYT @wooorm ?

**Comment by wooorm** [comment]
This is really good! I merged it, and changed it a bit (see da61635).
I did switch from `undefined` -> `defined`, and added both to the prototype.
I also made `data` props defined. As they are defined, it just made more sense to me.
I do see some value in adding a `known` prop to only include the spaces and aria, but not data or unlisted.

Or is this fine now? What do you think?

**Comment by StarpTech** [comment]
> I also made data props defined. As they are defined, it just made more sense to me.

This is what I want to avoid because I'm looking for a way to differentiate between custom properties like `foo` (non-spec) and `data-foo`. In my formatter I would like to allow to format attributes without quotes when it's a non-spec attribute.

**Comment by wooorm** [comment]
Wait. `foo` has `info.defined` set to `false`, whereas `data-foo` has `info.defined` set to `true`.

`data-foo *is* specced. 🤷‍♂️

**Comment by StarpTech** [comment]
You're right! I misunderstood it 😄 👍

**Comment by wooorm** [comment]
Yeah, just wanted to check. In your PR they got `defined: false`.

Ok, so can I release this, or do you want something else in there?

**Comment by StarpTech** [comment]
No, it's fine as long as I can differentiate between `foo` and (`data-*`, `disabled` ...)

**Comment by StarpTech** [comment]
You can release it 🌟

**Comment by wooorm** [comment]
Done! 4.2.0
